From 2c8948e76ee431c1df4b552aedf477ab65f98bb1 Mon Sep 17 00:00:00 2001
From: Eyal Reizer <eyalr@ti.com>
Date: Wed, 15 Apr 2015 16:37:30 +0300
Subject: [PATCH] mac80211: iface: adapt alloc_netdev_mqs

The prototype of the function was changed in 3.17 so we have to adapt
the driver for this

Signed-off-by: Eyal Reizer <eyalr@ti.com>
---
 net/mac80211/iface.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/mac80211/iface.c b/net/mac80211/iface.c
index 2a602dc..252b221 100644
--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -1626,9 +1626,15 @@ int ieee80211_if_add(struct ieee80211_local *local, const char *name,
 		if (local->hw.queues >= IEEE80211_NUM_ACS)
 			txqs = IEEE80211_NUM_ACS;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,17,0)
 		ndev = alloc_netdev_mqs(sizeof(*sdata) +
 					local->hw.vif_data_size,
 					name, ieee80211_if_setup, txqs, 1);
+#else
+		ndev = alloc_netdev_mqs(sizeof(*sdata) + local->hw.vif_data_size,
+					name, NET_NAME_UNKNOWN,
+					ieee80211_if_setup, txqs, 1);
+#endif
 		if (!ndev)
 			return -ENOMEM;
 		dev_net_set(ndev, wiphy_net(local->hw.wiphy));
-- 
1.7.9.5

